File: src/main/java/com/musinsa/subject/service/facade/PriceFacade.java
package com.musinsa.subject.service.facade;

import com.musinsa.subject.model.price.LowestPriceByCategoryResponse;
import com.musinsa.subject.model.price.LowestTotalPriceBrandWithCategoryDetailsResponse;
import com.musinsa.subject.model.price.MinMaxPriceBrandsByCategoryNameResponse;

public interface PriceFacade {

    LowestPriceByCategoryResponse getLowestPriceByCategory();

    LowestTotalPriceBrandWithCategoryDetailsResponse getLowestTotalPriceBrandWithCategoryDetails();

    MinMaxPriceBrandsByCategoryNameResponse getMinMaxPriceBrandsByCategoryName(String categoryName);

}

File: src/main/kotlin/com/musinsa/subject/service/facade/PriceFacade.kt
package com.musinsa.subject.service.facade

import com.musinsa.subject.model.price.LowestPriceByCategoryResponse
import com.musinsa.subject.model.price.LowestTotalPriceBrandWithCategoryDetailsResponse
import com.musinsa.subject.model.price.MinMaxPriceBrandsByCategoryNameResponse

interface PriceFacade {
    fun getLowestPriceByCategory(): LowestPriceByCategoryResponse
    fun getLowestTotalPriceBrandWithCategoryDetails(): LowestTotalPriceBrandWithCategoryDetailsResponse
    fun getMinMaxPriceBrandsByCategoryName(categoryName: String): MinMaxPriceBrandsByCategoryNameResponse
}
